sklearn/tests/test_metadata_routing.py

```diff
@@ -73,14 +73,16 @@ def assert_request_equal(request, dictionary):
 def record_metadata(obj, method, **kwargs):
     """Utility function to store passed metadata to a method."""
     if not hasattr(obj, "_records"):
-        setattr(obj, "_records", dict())
+        obj._records = {}
     obj._records[method] = kwargs
 
 
 def check_recorded_metadata(obj, method, **kwargs):
     """Check whether the expected metadata is passed to the object's method."""
     records = getattr(obj, "_records", dict()).get(method, dict())
-    assert set(kwargs.keys()) == set(records.keys())
+    # only check keys whose value was explicitly passed
+    expected_keys = {key for key, val in records.items() if val is not None}
+    assert set(kwargs.keys()) == expected_keys
```

Comment:
Thinking more about this, the change is making the test weaker, and it makes it weaker everywhere.

I think a safer option would be to leave this function as is, and change `record_metadata` to accept a `record_none` arg, which is `True` by default, and in our Consumer estimators in this PR we can set that arg to `False`.

Reply:
Makes sense, I changed it as you suggested.